readthedocs/search/models.py

```diff
@@ -137,3 +137,60 @@ def generate_distribution_of_top_queries(cls, project_slug, n):
             final_data['int_data'].append(count_of_other)
 
         return final_data
+
+
+class PageView(TimeStampedModel):
+    project = models.ForeignKey(
+        Project,
+        related_name='page_views',
+        on_delete=models.CASCADE,
+    )
+    version = models.ForeignKey(
+        Version,
+        verbose_name=_('Version'),
+        related_name='page_views',
+        on_delete=models.CASCADE,
+    )
+    path = models.CharField(max_length=4096)
+    view_count = models.PositiveIntegerField(default=1)
+
+    def __str__(self):
+        return f'[{self.project.slug}:{self.version.slug}]: {self.path}'
+
+    @classmethod
+    def get_top_viewed_pages(cls, project):
+        """
+        Returns top 10 pages according to view counts.
+
+        Structure of returned data is compatible to make graphs.
+        Sample returned data::
+        {
+            'pages': ['index.html', 'contribute.html', 'sponsors.html'],
+            'view_counts': [150, 200, 143]
+        }
+        This data shows that `index.html` is the most viewed page having 150 total views,
+        followed by `contribute.html` and `sponsors.html` having 200 and
+        143 total page views respectively.
+        """
+        qs = (
+            cls.objects
+            .filter(project=project)
+            .values_list('path')
+            .annotate(total_views=Sum('view_count'))
+            .values_list('path', 'total_views')
+            .order_by('-total_views')[:10]
+        )
```

Comment:
I'm pretty worried about this query. It seems like we're going to be doing something that is definitely not indexed in the DB across a whole lot of fields. 

Can you legally have two `values_list` calls? How does that work?

Reply:
I think it is correct.
This generates the following SQL query.

```sql
SELECT "search_pageview"."path", SUM("search_pageview"."view_count") AS "total_views" 
FROM "search_pageview"
WHERE "search_pageview"."project_id" = 62
GROUP BY "search_pageview"."path"
ORDER BY "total_views" DESC
LIMIT 10
```

Reply:
This query is probably ok since it's limited by the project.